ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/qos_profiles.schema.yml

```
@@ -78,6 +78,39 @@ keys:
                     description: |
                       Supported options are platform dependent
                       Example: "< rate > kbps", "1-100 percent", "< rate > pps"
+              random_detect:
+                type: dict
+                keys:
+                  ecn:
+                    type: dict
+                    keys:
+                      count:
+                        type: bool
+                        description: |
+                          Turn on random-detect ecn count
+                      threshold:
+                        type: dict
+                        keys:
+                          units:
+                            type: str
+                            valid_values: ["segments", "bytes", "kbytes", "mbytes"]
+                          min:
+                            type: int
+                            description: |
+                              Set the random-detect ecn minimum-threshold
```

No need to repeat the key names etc so just try to help the user fill in expected values. Like for various timers we say "timeout in seconds" etc.
Same for the other keys in this area.

```suggestion
                              Minimum threshold in ???unit???
```

---

After looking through the templates, I know understand that this is your own key - not part of the CLI as such. Is is a requirement that all values are using the same unit or would some users want different units?

---

The EOS command accepts an integer value and a unit for each of minimum-threshold and maximum-threshold. There are multiple units supported,  but I could not find any documentation whether the same command (for the same queue) can have mixed units. So here I am assuming that all values for 1 queue will be using the same unit.

---

I test on a `DCS-7050DX4-32S-F` and the units used in the same command/line need to match. So I believe we have the right setup here, where only 1 unit can be specified for a single queue.

---

OK, thank you for testing and confirming. Then just make the description `Minimum threshold` etc. and we keep the units like you have it.